### ISSUE TITLE
Syncoid data transfer speed goes to 0 after 660gb transferred.

### ISSUE THREAD

**Issue by joe-eklund** [jimsalterjrs/sanoid#590]
Hello-

I have two machines, primary and secondary (PM and SM). These two machines are connected over a gigabit connection through the same ISP. I would like to use syncoid to transfer all the PM snapshots to SM. 

To accomplish this I ran this command on the PM: `syncoid --no-privilege-elevation --debug zfs_pool/pool joe@osgiliath:citadel/pool`.

It started out just fine going at my max upload speed (roughly 800-900 mbps). Then at around 660 GB the network traffic stopped and it just says transferring at `[0.00 B/s]`. I can `ctrl-c` out of the process, then I can't start it up again as it says: `Cannot sync now: citadel/pool is already target of a zfs receive process.`. I then tried killing the receive process on SM. It didn't work so I rebooted the computer, and it eventually timed out the zfs recv command.

I was then able to start the syncoid command again. It stopped again around 660 GB. 

Rinse and repeat. Any ideas why it's stopping?

I am moving over from znapzend because of other issues, and would love to use syncoid if I could get it to work.

**Comment by jimsalterjrs** [comment]
This could be a network issue, or could be a storage issue on the target. You'll need to troubleshoot further to pinpoint the actual issue. Syncoid itself is simply orchestrating ZFS replication; if the process *begins* there's not really anything to troubleshoot within syncoid itself.

You might want to try using `zpool iostat` on the target and seeing what the %util and actual write in MiB/sec looks like on the target pool. If you don't have any activity there, you've got a network issue. If you *do* have a bunch of writes there, you're looking at the storage not being able to keep up, and you'll need to figure out why it can't.

**Comment by phreaker0** [comment]
@joe-eklund like @jimsalterjrs already said, this has nothing to do with syncoid but with the zfs send/recv. There is probably an issue with your pool (look for zfs messages in the kernel log) or it's just really slow at this point because of huge amount of metadata operations (happened to me too on an all hdd pool where I deleted millions of small files).

**Comment by joe-eklund** [comment]
Just wanted to report back that I was able to get my ~100 TB of data syncoided over! I think a combination of following things I did fixed my issue:

- Reinstalled the OS on my target. It was having problems rebooting correctly. So that got rid of those lockups and errors.
- Completely remade the target ZFS pool.
- I had two drives get marked as faulted on the source system and replaced them. This drastically reduced the errors and I only had to restart the transfer a couple times. Subsequent transfers (that are much smaller) have had no problem going through.

I suspect it was the dying drives that were the main problem. I also noticed another drive on the source has increasing reallocated sectors, so that may have also contributed to the few times I did have to restart the transfer (after my initial two dying hard drives were already replaced).

Just need to setup it to automatically sync, but now that my initial syncoid command went through I am very happy. Thank you @jimsalterjrs for making syncoid/sanoid! They worked great once I fixed my underlying issues. :)
